code/game/objects/items/devices/spyglasses.dm

```diff
@@ -0,0 +1,91 @@
+//antag spyglasses. meant to be an example for map_popups.dm
+/obj/item/clothing/glasses/regular/spy
+	desc = "Made by Nerd. Co's infiltration and surveillance departmentt. Upon closer inspection, there's a small screen put into the lenses."
+	var/obj/item/spy_bug/linked_bug
+
+/obj/item/clothing/glasses/regular/spy/proc/show_to_user(var/mob/user)//this is the meat of it. most of the map_popup usage is in this.
+	if(!user)
+		return
+	if(!user.client)
+		return	
+	if(!linked_bug)
+		user.audible_message("<span class='warning'>[src] lets off a shrill beep!</span>")
+
+	if("spypopup_map" in user.client.screen_maps) //alright, the popup this object uses is already IN use, so the window is open. no point in doing any other work here, so we're good. 
+		return 
+	user.client.setup_popup("spypopup",3,3,2)
+	var/list/buglist = list(linked_bug.cam_view,linked_bug.popupmaster)
+	user.client.add_objs_to_map(buglist)
+	linked_bug.update_view()
+
+/obj/item/clothing/glasses/regular/spy/dropped(mob/user)
+	. = ..()
+	user.client.close_popup("spypopup_map")
+
+/obj/item/clothing/glasses/regular/spy/verb/activate_remote_view()
+	//yada yada check to see if the glasses are in their eye slot
+	show_to_user(usr)
+
+/obj/item/spy_bug
+	name = "pocket protector"
+	icon = 'icons/obj/clothing/accessories.dmi'
+	icon_state = "pocketprotector"
+	desc = "an advanced holographic projection in the shape of a pocket protector featuring technology similar to a chameleon projector. it has a built in 360 degree camera for all your nefarious needs. Simply hitting an object with it will cause it's projection to change. Microphone not included."
+	var/obj/item/clothing/glasses/regular/spy/linked_glasses
+	var/obj/screen/cam_view
+	var/obj/screen/plane_master/lighting/popupmaster 
```

Comment:
what is this even used for

Reply:
plane masters only apply to screen they're on

Reply:
so the lighting overlays wouldn't apply otherwise?

i don't even understand why there would be a lighting overlay if it's using a screen object

Reply:
@optimumtact

Plane masters only apply to the map their screen loc is set to (normally it's just the main map, which is fine - I'm pretty sure the only other map in the game thus far is the character create)

However, this becomes an issue in this specific circumstance, as when turfs are added to vis contents, **everything** on the turf is rendered along with it, *including lighting overlays*.

This results in the overlay failing to blend due to a lack of a planemaster, which looks something like [this](https://media.discordapp.net/attachments/484170915253321734/634803956241006592/image.png)

That is exactly how it'd look on the main map if the planemaster was deleted.

Reply:
yeah anturk explained in discord but thanks as well